### PR TITLE
Add support for links that targets directly to the latest version

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR fixes an issue where links that point explicitly to the latest version did not work (e.g. https://dev.handsontable.com/docs/9.0/license-key/). I tried to find some built-in Vuepress options that would cover the issue but I didn't found any. 

~To make sure that the redirection is working in build mode it is necessary to generate routes for all versions, even for the latest. The latest version then is used to set redirection to the path that does not contain the version in itself.~

~As each version has generated its own routes (e.g. `/9.0/license-key/`, `/8.4/license-key/`) it's necessary to add a new directory with a version that keeps the content of the latest version. I used symlink to do that. Thanks to that the route is always in sync with the latest version. That files are [processed differently](https://github.com/handsontable/handsontable/pull/8336/files#diff-52eb86351480925d61947a58caca1ea76d10950ec26533f16f37cb12a3519dffR104) to achieve working docs.~

This is the second iteration of the problem-solving approach. In that changes, I pass the redirection responsibility to the server-side (Nginx). While building the docs the rewrite rules are generated in the `docker/redirects.conf` file. That file is then copied to the production Docker image and becomes part of the Nginx configuration. Thanks to that all links that directly point to the latest version are instantly redirected (with 301 HTTP code) to the proper page.

As the `config.js` file is getting bigger and bigger I decided to create a plugin (following the best practices from the VuePress). It should help resolve problems with conflicts where developers change the same file and should improve the codebase maintainability - a good start for improvements.

The redirection works only for docker builds.

### How has this been tested?

I tested on local docker build with another container that acts as a reverse proxy.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #8319
2.
3.

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by Siemienik** [comment]
did you attempted to use [`additionalPages`](https://vuepress.vuejs.org/plugin/option-api.html#additionalpages) setting?
I suppose this could work without any filesystem linking

**Comment by budnix** [comment]
@Siemienik Thankfully the symlinking turns out to be not necessary. The PR is ready for review.

**Comment by budnix** [comment]
@Siemienik My latest proposed solution does not work in the build mode. I think that the original solution has to be restored.

**Comment by budnix** [comment]
@Siemienik The PR is ready for review.

File: docs/.gitignore [hunk 1]
```diff
@@ -18,6 +18,7 @@ report-check-links.xlsx
 /.vuepress/public/handsontable.css
 
 /next/api/
+/docker/redirects.conf
```
Comment (Siemienik) [inline docs/.gitignore:21]: This line was added into the middle `/next/api` ignore defs:
```
/next/api/
!/next/api/introduction.md
!/next/api/plugins.md
!/next/api/sidebar.js
```

I want to keep it in one piece. Could You please move this line above or after ?

Comment (budnix) [reply docs/.gitignore:20]: Done https://github.com/handsontable/handsontable/pull/8336/commits/1f604a8f1975eb7a1a201408f60f95f89c76f973

**Review by Siemienik — APPROVED** [review]
LGTM 🥇 I tested it locally and on the stage environment, Both works perfectly!

**Pull request merged**